src/create_wand_data.cpp

```diff
@@ -42,20 +45,36 @@ int main(int argc, const char **argv) {
         (p_type == partition_type::fixed_blocks) ? "static partition" : "variable partition";
     spdlog::info("Block based wand creation with {}", partition_type_name);
 
-    binary_collection      sizes_coll((input_basename + ".sizes").c_str());
+    binary_collection sizes_coll((input_basename + ".sizes").c_str());
     binary_freq_collection coll(input_basename.c_str());
 
+    // Initialize the variant to the correct type
+    boost::variant<float, uint64_t> block_size = uint64_t(0);
```

Comment:
We probably want to make it `const`, so I'd either use another function to set this up, or just wrap it in a lambda:

```cpp
auto const block_size = [&] () -> boost::variant<float, uint64_t> {
    if (variable_block) {
        return lambda;
    } else {
        return fixed_block_size;
    }
}();
```
Not that I moved checking compliance of defined options above to the CLI part.